libsimulator/src/SocialForceModel.cpp

```diff
@@ -0,0 +1,182 @@
+#include "SocialForceModel.hpp"
+
+#include "Ellipse.hpp"
+#include "GenericAgent.hpp"
+#include "Macros.hpp"
+#include "Mathematics.hpp"
+#include "NeighborhoodSearch.hpp"
+#include "OperationalModel.hpp"
+#include "OperationalModelType.hpp"
+#include "Simulation.hpp"
+#include "SocialForceModelData.hpp"
+
+#include <Logger.hpp>
+#include <iostream>
+#include <stdexcept>
+
+SocialForceModel::SocialForceModel(double bodyForce_, double friction_)
+    : bodyForce(bodyForce_), friction(friction_){};
+
+OperationalModelType SocialForceModel::Type() const
+{
+    return OperationalModelType::SOCIAL_FORCE;
+}
+
+std::unique_ptr<OperationalModel> SocialForceModel::Clone() const
+{
+    return std::make_unique<SocialForceModel>(*this);
+}
+
+OperationalModelUpdate SocialForceModel::ComputeNewPosition(
+    double dT,
+    const GenericAgent& ped,
+    const CollisionGeometry& geometry,
+    const NeighborhoodSearchType& neighborhoodSearch) const
+{
+    const auto& model = std::get<SocialForceModelData>(ped.model);
+    SocialForceModelUpdate update{};
+    auto forces = drivingForce(ped);
+
+    const auto neighborhood = neighborhoodSearch.GetNeighboringAgents(ped.pos, this->_cutOffRadius);
+    Point F_rep;
```

Comment:
F_rep is now used to accumulate the forces of the other agents. This sum is then divided by the mass of the agent and added to the total forces instead of dividing the individual forces directly and adding them to the total forces. This is done to ensure that the floating point error is as small as possible